src/dtclient/kubernetes_settings.go

```diff
@@ -231,21 +231,29 @@ func createBaseRequest(url, method, apiToken string, body io.Reader) (*http.Requ
 }
 
 func handleErrorArrayResponseFromAPI(response []byte, statusCode int) error {
-	var se []getSettingsErrorResponse
-	if err := json.Unmarshal(response, &se); err != nil {
-		return fmt.Errorf("response error: %d, can't unmarshal json response: %w", statusCode, err)
-	}
+	if statusCode == http.StatusForbidden || statusCode == http.StatusUnauthorized {
+		var se getSettingsErrorResponse
```

Comment:
with the additional checks we hopefully shouldn't get any more forbidden/unauthorized responses, because it should fail before even attempting the request
let's leave it in for now. in the future we could handle responses with these 2 status codes in a reusable way